### PR TITLE
Adding docker and bosh manager jobs

### PR DESCRIPTION
**Pull request by subhankarc**
* New jobs for BOSH manager and Docker Manager
* settings are trimmed down for Docker
* Trimming down of the settings could not be done for BOSH, full settings is being put.

### CODE REVIEW

File: jobs/service-fabrik-bosh-manager/templates/config/settings.yml.erb [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+  ##############################
+  # EXTERNAL ENDPOINT SETTINGS #
+  ##############################
+  external:
+    log_event: <%= link("broker").p('external.log_event') %>
+    event_type: SF.API_EVENT
+    trust_proxy: <%= external['trust_proxy'] %>
+    port: <%= external['port'] %>
+    <% link("broker").if_p('external.ssl') do |ssl| %>
+    protocol: 'https'
```
Comment (sauravmndl) [inline jobs/service-fabrik-bosh-manager/templates/config/settings.yml.erb]: Can you put like this?
   protocol: 'https'
   <% link("broker").if_p('external.ssl') do |ssl| %>
    ssl: <%= JSON.dump(ssl) %>
    <% end %> 

else part not required.

File: jobs/service-fabrik-bosh-manager/templates/config/settings.yml.erb [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+    log_event: <%= link("broker").p('internal.log_event') %>
+    event_type: SF.BROKER_EVENT
+    domain_socket:
+      path: <%= link("broker").p('internal.domain_socket.path') %>
+      fwd_events: true
+    port: <%= internal['port'] %>
+    ip: <%= internal['ip'] %>
+    host: <%= "#{internal['ip']}:#{internal['port']}" %>
+    <% link("broker").if_p('internal.ssl') do |ssl| %>
+    protocol: 'https'
```
Comment (sauravmndl) [inline jobs/service-fabrik-bosh-manager/templates/config/settings.yml.erb]: same as above.

File: jobs/service-fabrik-bosh-manager/templates/bin/service-fabrik-bosh-manager_ctl.erb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      --startas ${NODE_CMD} \
+      --chuid vcap:vcap \
+      --background \
+      --no-close \
+      --chdir ${PACKAGE_DIR} \
+      -- ${PACKAGE_DIR}/managers/StartBoshManagers.js \
+      1>> ${LOG_DIR}/${OUTPUT_LABEL}.stdout.log \
+      2>> ${LOG_DIR}/${OUTPUT_LABEL}.stderr.log
+    ;;
+
```
Comment (sauravmndl) [inline jobs/service-fabrik-bosh-manager/templates/bin/service-fabrik-bosh-manager_ctl.erb]: Remove extra lines

File: jobs/service-fabrik-bosh-manager/templates/config/settings.yml.erb [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+  def ostruct_to_hash(os)
+    os.marshal_dump.map do |key, value|
+      { key => value.is_a?(OpenStruct) ? ostruct_to_hash(value) : value }
+    end.reduce(&:merge)
+  end
+
+  broker_name = link("broker").p('name')
+  log_path = "/var/vcap/sys/log/service-fabrik-bosh-manager/service-fabrik-bosh-manager.log"
+  networks = ostruct_to_hash(spec.networks)
+  default_ip = networks.values.find { |net| net.has_key?(:default) }[:ip]
```
Comment (sauravmndl) [inline jobs/service-fabrik-bosh-manager/templates/config/settings.yml.erb]: is it getting used anywhere?

File: jobs/service-fabrik-bosh-manager/templates/config/settings.yml.erb (hunk 1 above)
Comment (subhankarc) [reply jobs/service-fabrik-bosh-manager/templates/config/settings.yml.erb]: done.

File: jobs/service-fabrik-bosh-manager/templates/config/settings.yml.erb (hunk 2 above)
Comment (subhankarc) [reply jobs/service-fabrik-bosh-manager/templates/config/settings.yml.erb]: done.

**Pull request merged**